content/topics/java-specific/streams/_index.md

```diff
@@ -0,0 +1,71 @@
+---
+content_type: topic
+ready: true
+title: Java Streams
+---
+
+Streams are one of the major features introduced in java 8, what do they do you ask, well in short they give data sources super powers. These streams should not be confused with File I/O streams which are a completly different thing.
+
+#### The features of Java stream are –
+
+- A stream is not a data structure instead it takes input from the Collections, Arrays or I/O channels.
+
+- Streams don’t change the original data structure, they only provide the result as per the pipelined methods.
+
+- Each intermediate operation is lazily executed and returns a stream as a result, hence various intermediate operations can be pipelined. Terminal operations mark the end of the stream and return the result.
+
+#### We have two types of operations with streams
+
+***1. Intermediate Operations***
+
+These are operations that result into other streams which means more processing can still be done on them, like:
+
+- map: The map method is used to returns a stream consisting of the results of applying the given function to the elements of this stream.
```

Comment:
I think the methods can be inside bacticks.
- `map` : The map method ... 

returns => return